webapp/src/main/java/com/box/l10n/mojito/service/delta/DeltaService.java

```diff
@@ -0,0 +1,192 @@
+        deltaResponseDTO.setDeltaMetadataDTO(deltaMetadataDTO);
+
+        return deltaResponseDTO;
+    }
+
+    /**
+     * Queries the database to identify all Text Unit Variants that are newer and different than the variants that were
+     * used for the latest PullRuns.
+     * <p>
+     * To note: for the results to be accurate, each PushRun ID provided should have been processed before it's
+     * equivalent PullRun ID.
+     * <p>
+     * Overall logic looks like this:
+     *      1. Using the push_run_ids, collect a list of text units that were associated with the PushRuns
+     *      2. Using the pull_run_ids, create a list of the “base” translations that were included for
+     *      those PullRuns and collect a list of relevant text unit variants that correspond to the text units
+     *      3. Compare the “base” text unit variants with the “latest” variants and create a list of results with locale
+     *      information as well as the type of “translation change” that is being represented in the output
+     *      (e.g.: “NEW_TRANSLATION” vs. “UPDATED_TRANSLATION”).
+     *      4. Filter out any variants that were contained in the previous PullRuns or that have the same contents
+     *      as before.
+     *
+     * @return The delta of text unit variants, their translations and corresponding metadata.
+     */
+    public DeltaResponseDTO getDeltasForRuns(Repository repository,
+                                             List<Locale> locales,
+                                             List<PushRun> pushRuns,
+                                             List<PullRun> pullRuns) {
+
+        List<Long> localeIds = locales == null ? Collections.emptyList() :
+                locales.stream().map(BaseEntity::getId).collect(Collectors.toList());
+        List<Long> pushRunIds = pushRuns == null ? Collections.emptyList() :
+                pushRuns.stream().map(BaseEntity::getId).collect(Collectors.toList());
+        List<Long> pullRunIds = pullRuns == null ? Collections.emptyList() :
+                pullRuns.stream().map(BaseEntity::getId).collect(Collectors.toList());
+
+        List<TextUnitVariantDelta> variants = tmTextUnitVariantRepository.findDeltasForRuns(
+                repository.getId(),
+                localeIds,
+                pushRunIds,
+                pullRunIds);
+
+        Map<String, DeltaLocaleDataDTO> deltaLocaleDataByBcp47Tags = getStringDeltaLocaleDataDTOMap(variants);
+
+        DeltaMetadataDTO deltaMetadataDTO = new DeltaMetadataDTO();
+
+        DeltaResponseDTO deltaResponseDTO = new DeltaResponseDTO();
+        deltaResponseDTO.setTranslationsPerLocale(deltaLocaleDataByBcp47Tags);
+        deltaResponseDTO.setDeltaMetadataDTO(deltaMetadataDTO);
+
+        return deltaResponseDTO;
+    }
+
+    private TextUnitVariantDelta getTextUnitVariantDelta(TMTextUnitVariant tmTextUnitVariant) {
+        return new TextUnitVariantDelta() {
```

Comment:
might be better to have an actual implementation vs the annonymous class

Reply:
Done